### PR TITLE
[develop] Remove creation of EFS mount targets for external FS

### PR DESCRIPTION
**Pull request by lukeseawalker**
Signed-off-by: Luca Carrogu <carrogu@amazon.com>

### Description of changes
Before this patch PCluster create EFS mount targets for existing FS if there isn't one already in the AZs of the cluster. Such mount target resource is therefore associated with a cluster lifecycle and is deleted when the cluster is deleted.

Due to the fact that each EFS can have a single mount target per AZ, in case the user mounts the same FS in multiple clusters, when the first one is deleted all other clusters will see the mount target revoked and will break.

Following the same approach for all un-managed PCluster resources, within this patch it is removed the creation of missing mount target for existing EFS and simply fails at validation time when requirements are not met.

### Tests
* added unit tests

### References
* n/a

### Checklist
- [X] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [X] Check all commits' messages are clear, describing what and why vs how.
- [X] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [X] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/validators/test_cluster_validators.py [hunk 1]
```diff
@@ -1272,8 +1272,41 @@ def test_deletion_policy_validator(deletion_policy, name, expected_message, fail
 
 @pytest.mark.parametrize(
     "avail_zones_mapping, cluster_subnet_cidr, are_all_security_groups_customized, security_groups, expected_message, "
-    "failure_level",
+    "failure_level, file_system_info",
```
Comment (EddyMM) [inline cli/tests/pcluster/validators/test_cluster_validators.py]: NIT: The expected message/failure_level should be the last parameter (based on the structure of other tests)

File: cli/src/pcluster/aws/efs.py [hunk 2]
```diff
@@ -55,3 +55,28 @@ def describe_mount_targets(self, efs_fs_id):
...
+        availability_zone_name = (
+            self.describe_file_system(efs_fs_id).get("FileSystems")[0].get("AvailabilityZoneName", None)
+        )
+        return availability_zone_name is None
+
+    @AWSExceptionHandler.handle_client_exception
+    @Cache.cached
+    def describe_file_system(self, efs_fs_id):
+        """
+        Decsribe file system for the given EFS file system id.
```
Comment (EddyMM) [inline cli/src/pcluster/aws/efs.py]: NIT: typo

Comment (lukeseawalker) [reply cli/src/pcluster/aws/efs.py]: fixed

File: cli/tests/pcluster/validators/test_cluster_validators.py (hunk 1 above)
Comment (lukeseawalker) [reply cli/tests/pcluster/validators/test_cluster_validators.py]: moved

**Review by EddyMM — APPROVED** [review]

File: cli/src/pcluster/aws/efs.py [hunk 3]
```diff
@@ -55,3 +55,28 @@ def describe_mount_targets(self, efs_fs_id):
...
+    @Cache.cached
+    def is_efs_standard(self, efs_fs_id):
+        """
+        Return true if EFS storage class is Standard, otherwise false (OneZone).
+
+        :param efs_fs_id: EFS file system Id
+        :return: true if EFS storage class is Standard
+        """
+        availability_zone_name = (
+            self.describe_file_system(efs_fs_id).get("FileSystems")[0].get("AvailabilityZoneName", None)
```
Comment (hanwen-pcluste) [inline cli/src/pcluster/aws/efs.py:69]: Minor the `None` in `("AvailabilityZoneName", None)` is optional

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**